provider/core/src/lib.rs

```diff
@@ -154,16 +154,15 @@ pub mod serde;
 // Types from private modules
 pub use crate::data_provider::BoundDataProvider;
 pub use crate::data_provider::DataProvider;
-pub use crate::data_provider::DataProviderWithKey;
+pub use crate::data_provider::DataProviderWithMarker;
 pub use crate::data_provider::DynamicDataProvider;
 pub use crate::error::DataError;
 pub use crate::error::DataErrorKind;
-pub use crate::key::DataKey;
-pub use crate::key::DataKeyHash;
-pub use crate::key::DataKeyMetadata;
-pub use crate::key::DataKeyPath;
-pub use crate::request::DataKeyAttributes;
+pub use crate::key::DataMarkerInfo;
+pub use crate::key::DataMarkerPath;
```

Comment:
Should DataMarkerPath go in the prelude?

Reply:
I think that's something we can consider later, if we can change the datagen API to accept paths instead of infos.

Reply:
Seems like we should have DataMarkerPath, DataMarkerPath+DataMarkerPathHash, or neither? Right now we have DataMarkerPathHash but not DataMarkerPath in the prelude

Reply:
I'll remove the hash for now, I think that's specific enough to not be in the prelude